types/src/transaction/mod.rs

```diff
@@ -233,9 +269,36 @@ impl RawTransaction {
         public_key: Ed25519PublicKey,
     ) -> Result<SignatureCheckedTransaction> {
         let signature = private_key.sign(&self);
-        Ok(SignatureCheckedTransaction(SignedTransaction::new(
-            self, public_key, signature,
-        )))
+        Ok(SignatureCheckedTransaction(
+            SignedTransaction::new(
+                self,
+                AccountAuthenticator::ed25519(public_key, signature),
+            ),
+        ))
+    }
+
+    pub fn sign_multi_agent(
+        self,
+        sender_private_key: &Ed25519PrivateKey,
+        sender_public_key: Ed25519PublicKey,
+        secondary_private_keys: Vec<&Ed25519PrivateKey>,
+        secondary_public_keys: Vec<Ed25519PublicKey>
```

Comment:
Probably want to add a comment here explaining how these keys need to match the exact order/length of the `secondary_signers` field in the RawTransaction